src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```
@@ -101,102 +111,95 @@ trait SharedFileSystem {
+  def adjustInputPaths(callKey: CallKey, inputs: CallInputs, workflowDescriptor: WorkflowDescriptor): CallInputs = {
+    val call = callKey.scope
+    val strategies = if (call.docker.isDefined) DockerLocalizers else Localizers
+
+    def toDockerPath(path: Path): Path = {
       // Host path would look like cromwell-executions/three-step/f00ba4/call-ps/stdout.txt
       // Container path should look like /root/f00ba4/call-ps/stdout.txt
-      val fullPath = Paths.get(path).toFile.getAbsolutePath
+      val fullPath = path.toFile.getAbsolutePath
       // Strip out everything before cromwell-executions.
       val pathUnderCromwellExecutions = fullPath.substring(fullPath.indexOf(CromwellExecutionRoot) + CromwellExecutionRoot.length)
       // Strip out the workflow name (the first component under cromwell-executions).
       val pathWithWorkflowName = Paths.get(pathUnderCromwellExecutions)
-      WdlFile(WdlFile.appendPathsWithSlashSeparators("/root", pathWithWorkflowName.subpath(1, pathWithWorkflowName.getNameCount).toString))
+      Paths.get(WdlFile.appendPathsWithSlashSeparators("/root", pathWithWorkflowName.subpath(1, pathWithWorkflowName.getNameCount).toString))
     }
 
-    // If this call is using Docker, adjust input paths, otherwise return unaltered input paths.
-    def adjustPath(nameAndValue: (String, WdlValue)): (String, WdlValue) = {
-      val (name, value) = nameAndValue
-      val adjusted = value match {
-        case WdlFile(path) => containerPath(path)
-        case WdlArray(t, values) => new WdlArray(t, values map { adjustPath(name, _)._2 })
-        case WdlMap(t, values) => new WdlMap(t, values mapValues { adjustPath(name, _)._2 })
-        case x => x
-      }
-      name -> adjusted
+    /**
+     * Transform an original input path to a path in the call directory.
+     * The new path matches the original path, it only "moves" the root to be the call directory.
+     */
+    def toCallPath(path: Path): Try[Path] = Try {
```

Is that because it can't fail or some sort of "let it fail" angle?